lib/dialects/abstract/connection-manager.js

```diff
@@ -295,6 +295,11 @@ class ConnectionManager {
   releaseConnection(connection) {
     return this.pool.release(connection).tap(() => {
       debug('connection released');
+    }).catch((e) => {
+      // ISSUE(7625) No-op if generic-pool is unable to release the connection, it may have already been released.
+      if (!/Resource not currently part of this pool/.exec(e.message)) {
```

Comment:
Is it possible to check if the resource is defined in the pool before calling destroy?